File: app/src/test/java/AddExpenseActivity.java
package com.example.simplebudget;

import android.content.Intent;
import android.os.Bundle;
import android.widget.Button;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;

public class DashboardActivity extends AppCompatActivity {

    TextView welcomeText;
    Button addExpenseBtn, searchBtn;
    String username;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_dashboard);

        welcomeText = findViewById(R.id.welcomeText);
        addExpenseBtn = findViewById(R.id.addExpenseBtn);
        searchBtn = findViewById(R.id.searchBtn);

        username = getIntent().getStringExtra("username");
        welcomeText.setText("Welcome, " + username);

        addExpenseBtn.setOnClickListener(v -> {
            Intent intent = new Intent(getApplicationContext(), AddExpenseActivity.class);
            intent.putExtra("username", username);
            startActivity(intent);
        });

        searchBtn.setOnClickListener(v -> {
            Intent intent = new Intent(getApplicationContext(), SearchActivity.class);
            intent.putExtra("username", username);
            startActivity(intent);
        });
    }
}

File: app/src/main/java/com/example/davisbudgetapp/AddExpenseActivity.kt
package com.example.budgetapp

import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import androidx.activity.viewModels
import androidx.appcompat.app.AppCompatActivity
import com.example.budgetapp.viewmodel.ExpenseViewModel

class AddExpenseActivity : AppCompatActivity() {

    private val viewModel: ExpenseViewModel by viewModels()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_add_expense)

        val title = findViewById<EditText>(R.id.titleInput)
        val amount = findViewById<EditText>(R.id.amountInput)
        val saveBtn = findViewById<Button>(R.id.saveBtn)

        saveBtn.setOnClickListener {
            val titleText = title.text.toString()
            val amountValue = amount.text.toString().toDoubleOrNull() ?: 0.0
            viewModel.addExpense(titleText, amountValue)
            finish()
        }
    }
}
